x-pack/plugins/apm/server/lib/transaction_groups/fetcher.ts

```diff
@@ -36,66 +38,159 @@ interface TopTraceOptions {
 export type Options = TopTransactionOptions | TopTraceOptions;
 
 export type ESResponse = PromiseReturnType<typeof transactionGroupsFetcher>;
+
+export type TransactionGroupRequestBase = ESSearchRequest & {
+  body: {
+    aggs: {
+      transaction_groups: Unionize<
+        Pick<AggregationOptionsByType, 'composite' | 'terms'>
+      >;
+    };
+  };
+};
+
+export type TransactionGroupSetup = Setup & SetupTimeRange & SetupUIFilters;
+
 export async function transactionGroupsFetcher(
   options: Options,
-  setup: Setup & SetupTimeRange & SetupUIFilters,
+  setup: TransactionGroupSetup,
   bucketSize: number
 ) {
-  const { client } = setup;
-
   const projection = getTransactionGroupsProjection({
     setup,
     options,
   });
 
-  const sort: SortOptions = [
-    { _score: 'desc' as const }, // sort by _score to ensure that buckets with sampled:true ends up on top
-    { '@timestamp': { order: 'desc' as const } },
-  ];
-
   const isTopTraces = options.type === 'top_traces';
 
-  if (isTopTraces) {
-    // Delete the projection aggregation when searching for traces, as it should use the combined aggregation instead
-    delete projection.body.aggs;
-  }
+  delete projection.body.aggs;
+
+  const size = bucketSize + 1;
 
-  const params = mergeProjection(projection, {
+  const request = mergeProjection(projection, {
```

Comment:
I keep running into the same issue: when I try to review these changes I end up in the rabbit hole of projections. 
I started flattening it back into a single file (mostly) and I think it really improves the readability and makes it easier to spot problems.

I think it might be a good idea to go back to the drawing board and find a way to simplify ui filters (and thus projections).

Reply:
My concern is that we're forgetting why we have projections, and end up duplicating large parts of our queries in separate files. Tbh, I don't think projections are the cause of - or the solution to - some of our issues. But curious to see what you come up with.

Reply:
>  I don't think projections are the cause of - or the solution to - some of our issues

Projections most definitely make it harder for me to make sense of the queries. But I'm not sure my suggestions are any better (I obviously think they are but I also know that most thinks sound better before they are converted to code).